src/sentry/integrations/slack/unfurl/discover.py

```diff
@@ -161,6 +161,23 @@ def unfurl_discover(
         if "daily" in display_mode:
             params.setlist("interval", ["1d"])
 
+        if "bar" in display_mode:
+            interval_is_valid = False
+            if "statsPeriod" in params:
+                delta = parse_stats_period(params.get("statsPeriod"))
+                if delta is not None:
+                    interval = get_interval_from_range(delta, False)
+                    interval_is_valid = True
+            else:
+                start, end = parse_timestamp(params.get("start")), parse_timestamp(
+                    params.get("end")
+                )
+                if start is not None and end is not None:
+                    interval = get_interval_from_range(end - start, False)
+                    interval_is_valid = True
+            if interval_is_valid is True:
+                params.setlist("interval", [interval])
```

Comment:
The logic looks great! Suggesting some changes so we have fewer lines of code and add a default interval
```suggestion
        if "bar" in display_mode:
            interval = "1h"
            if "statsPeriod" in params:
                delta = parse_stats_period(params["statsPeriod"])
                if delta:
                    interval = get_interval_from_range(delta, False)
            elif "start" in params and "end" in params:
                start, end = parse_timestamp(params["start"]), parse_timestamp(params["end"])
                if start and end:
                    interval = get_interval_from_range(end - start, False)
            params.setlist("interval", [interval])
```